packages/core/accounts/kernel/createKernelAccount.ts

```diff
@@ -49,15 +49,20 @@ import { encodeDeployCallData as encodeDeployCallDataV07 } from "./utils/account
 import { accountMetadata } from "./utils/common/accountMetadata.js"
 import { eip712WrapHash } from "./utils/common/eip712WrapHash.js"
 
+type KernelSmartAccountNonceExtension = {
+    getNonce: (customNonceKey?: bigint) => Promise<bigint>
```

Comment:
I think a separate type isn't needed for this function sig. Can be just part of:
```
{
        kernelPluginManager: KernelPluginManager<entryPoint>
        generateInitCode: () => Promise<Hex>
        encodeCallData: (args: KernelEncodeCallDataArgs) => Promise<Hex>
    }